src/frame/endpoints/tests/UserApiEndpoints.test.ts

```diff
@@ -184,6 +184,36 @@ describe("UserApiEndpoints", () => {
         });
     });
 
+    describe("callUserKeyUpdateApi", () => {
+        it("calls Api with rotated private key and augmentation factor when private key rotation is requested", () => {
+            (ApiRequest.makeAuthorizedApiRequest as jasmine.Spy).and.returnValue(
+                Future.of({
+                    id: "user-10",
+                    foo: "bar",
+                })
+            );
+
+            ApiState.setCurrentUser(TestUtils.getFullUser());
+            ApiState.setDeviceAndSigningKeys(TestUtils.getEmptyKeyPair(), TestUtils.getSigningKeyPair());
+
+            UserApiEndpoints.callUserKeyUpdateApi(new Uint8Array([98, 103, 110]), new Uint8Array([99, 104, 111])).engage(
+                (e) => fail(e.message),
+                (userKeys: any) => {
+                    expect(userKeys).toEqual({
+                        id: "user-10",
+                        foo: "bar",
+                    });
+                    expect(ApiRequest.makeAuthorizedApiRequest).toHaveBeenCalledWith("users/user-10/keys/1", expect.any(Number), expect.any(Object)); // keyId?
+                    const request = (ApiRequest.makeAuthorizedApiRequest as jasmine.Spy).calls.argsFor(0)[2];
+                    expect(JSON.parse(request.body)).toEqual({
+                        userPrivateKey: expect.any(Object),
+                        augmentationFactor: expect.any(Object),
```

Comment:
You should be able to assert directly about these values since you provided them up above on the `callUserKeyUpdateApi`. After you address my comments above these values will be base64 encoded strings.